From df6b4b3163932a7ce9b004fd4abd54b3570ae5e9 Mon Sep 17 00:00:00 2001
From: INTEL-CAMERA WIP <wip@localhost>
Date: Tue, 11 Aug 2026 14:24:49 +0200
Subject: [PATCH 3/5] BOZZA media: dt-bindings: Add GalaxyCore GC8034

NON INVIABILE: manutentore ed email sono segnaposto, manca il
Signed-off-by, binding non validato (dtschema assente).

Add device tree bindings for the GalaxyCore GC8034 8 Mpixel raw image
sensor, modelled on the already accepted galaxycore,gc05a2.yaml.
---
 .../bindings/media/i2c/galaxycore,gc8034.yaml | 114 ++++++++++++++++++
 1 file changed, 114 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/media/i2c/galaxycore,gc8034.yaml

diff --git a/Documentation/devicetree/bindings/media/i2c/galaxycore,gc8034.yaml b/Documentation/devicetree/bindings/media/i2c/galaxycore,gc8034.yaml
new file mode 100644
index 000000000..5f01c1707
--- /dev/null
+++ b/Documentation/devicetree/bindings/media/i2c/galaxycore,gc8034.yaml
@@ -0,0 +1,114 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/media/i2c/galaxycore,gc8034.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: GalaxyCore gc8034 8M Pixel MIPI CSI-2 sensor
+
+maintainers:
+  - TODO Nome Cognome <email@reale>
+
+description:
+  The gc8034 is a raw image sensor with a MIPI CSI-2 image data interface
+  and an I2C control bus. The output format is raw Bayer.
+
+properties:
+  compatible:
+    const: galaxycore,gc8034
+
+  reg:
+    maxItems: 1
+
+  clocks:
+    maxItems: 1
+
+  avdd-supply: true
+
+  dvdd-supply: true
+
+  dovdd-supply: true
+
+  reset-gpios:
+    description: Reference to the GPIO connected to the RESETB pin.
+    maxItems: 1
+
+  powerdown-gpios:
+    description: Reference to the GPIO connected to the PWDN pin.
+    maxItems: 1
+
+  port:
+    $ref: /schemas/graph.yaml#/$defs/port-base
+    additionalProperties: false
+    description:
+      Output port node, single endpoint describing the CSI-2 transmitter.
+
+    properties:
+      endpoint:
+        $ref: /schemas/media/video-interfaces.yaml#
+        unevaluatedProperties: false
+
+        properties:
+          data-lanes:
+            oneOf:
+              - items:
+                  - const: 1
+                  - const: 2
+                  - const: 3
+                  - const: 4
+              - items:
+                  - const: 1
+                  - const: 2
+
+          link-frequencies: true
+
+        required:
+          - data-lanes
+          - link-frequencies
+
+    required:
+      - endpoint
+
+required:
+  - compatible
+  - reg
+  - clocks
+  - avdd-supply
+  - dvdd-supply
+  - dovdd-supply
+  - reset-gpios
+  - port
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/gpio/gpio.h>
+
+    i2c {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        sensor@37 {
+            compatible = "galaxycore,gc8034";
+            reg = <0x37>;
+
+            clocks = <&gc8034_clk>;
+
+            reset-gpios = <&pio 21 GPIO_ACTIVE_LOW>;
+
+            avdd-supply = <&gc8034_avdd>;
+            dovdd-supply = <&gc8034_dovdd>;
+            dvdd-supply = <&gc8034_dvdd>;
+
+            port {
+                sensor_out: endpoint {
+                    data-lanes = <1 2 3 4>;
+                    link-frequencies = /bits/ 64 <336000000>;
+                    remote-endpoint = <&csi_in>;
+                };
+            };
+        };
+    };
+
+...
-- 
2.47.3

